src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -3941,7 +3941,20 @@ module TcDeclarations =
              | SynMemberDefn.NestedType (range=m) :: _ -> errorR(Error(FSComp.SR.tcTypesCannotContainNestedTypes(), m))
              | _ -> ()
         | ds ->
-            // Classic class construction 
+            // Check for duplicated parameters
+            ds
+            |> List.choose (function SynMemberDefn.AbstractSlot (x, _, m) -> Some(x, m) | _ -> None)
```

Comment:
Can `List.filter isAbstractSlot` be used here instead?

Reply:
I do not  think I can . Because there I need to get access to the he ```SynValSig``` to get the param names.  ```List.filter isAbstractSlot``` will give me back a boolean

Reply:
Filter will return all elements which satisfy predicates.

Reply:
Should be good now . Update the code to use your suggestion . Thanks